### DISCUSSION TITLE
No shutdown event (lifespan cleanup) when using --reload

### DISCUSSION THREAD

**Discussion by GuillaumeDesforges** [Kludex/uvicorn#1905] [Potential Issue] [answered]
It seems my lifespan cleanup actions are not triggered when I§ use uvicorn with `--reload`, I suspect shutdown event is not called.

Rough idea:

```python
@asynccontextmanager
async def lifespan(app: Starlette) -> AsyncGenerator[State, None]:
    print("START")
    yield None
    print("SHUTDOWN")

app = FastAPI(lifespan=lifespan)
```

Without `--reload` it prints both START and SHUTDOWN, but no if I use `--reload`.

**Comment by Kludex** [comment, marked as answer]
Are you on windows?

**Reply by GuillaumeDesforges** [reply]
Yes

**Reply by Kludex** [reply]
- Can you help to test https://github.com/encode/uvicorn/pull/1584 ?
